src/ChildProcess.php

```diff
@@ -12,20 +12,35 @@ class ChildProcess
 
     public function __construct(protected Client $client) {}
 
-    public function start(string $alias, array $cmd, ?string $cwd = null, ?array $env = null): object
+    public function start(string $alias, string|array $cmd, ?string $cwd = null, ?array $env = null): self
     {
         $this->alias = $alias;
 
+        $cwd = $cwd ?? base_path();
+
+        $cmd = is_iterable($cmd)
+            // when an array is passed, escape spaces for each item
+            ? array_map(fn ($a) => str_replace(' ', '\ ', $a), $cmd)
+            // when a string is passed, explode it on the space
+            : array_values(array_filter(explode(' ', $cmd)));
```

Comment:
I wanted to allow for passing a string as well as a array. There is a slight difference in behaviour here. I'm not quite happy with yet. 

Let me explain;

The request to the electron process expects the command to be an array. When passing a string, I want to explode it on a space and then send it with the request.

The problem arises when using a path containing spaces in your command.

For example: I have a space in the name of my home directory. The `PHP_BINARY` constant in in my dev environment equals `/Users/gwleuverink/Library/Application Support/Herd/bin/php83`.

When passing this constant inside a string I need the command to be posted like this:

``` php
[
    '/Users/gwleuverink/Library/Application Support/Herd/bin/php83',
    'artisan',
    'command:run'
]
```

Instead of:

``` php
[
    '/Users/gwleuverink/Library/Application',
    'Support/Herd/bin/php83',
    'artisan',
    'command:run'
]
```

Reply:
There also is the issue of escaping spaces. 

Right now I added this `str_replace(' ', '\ ', $a), $cmd)` to escape any space when the passed command was an array. I don't think we need this actually. 

People should be aware though that concatenating the `PHP_BINARY` constant inside a command string might be problematic. Since those strings are exploded on a space before being sent to Electron.

Reply:
Interesting! I hadn't seen this behaviour. `PHP_BINARY` isn't strictly the problem here, it's just down to how paths are being passed between the different APIs. I suspect rather than trying simply escape spaces, we should use [`escapeshellarg`](https://www.php.net/manual/en/function.escapeshellarg.php) instead

Reply:
I've removed the escaped spaces since it wasn't necessary. It seems electron already takes care of this. The problem only exists when passing the command as a string.

`ChildProcess::strart("PHP_BINARY artisan command:run")` is passed to the electron process like so:

``` php
[
    '/Users/gwleuverink/Library/Application',
    'Support/Herd/bin/php83',
    'artisan',
    'command:run'
]
```

Like you said in another comment; If we borrow this from Symfony process we might circumvent this issue all together. 

I'll resolve this conversation in favour of the other one